store/postgres/src/relational.rs

```diff
@@ -408,7 +416,9 @@ impl Layout {
                 "entity_count" => entity_count
             );
         }
-        let query = FilterQuery::new(&self, collection, filter.as_ref(), order, range, block)?;
+
+        let filter_collection = FilterCollection::new(&self, collection, filter.as_ref())?;
+        let query = FilterQuery::new(&filter_collection, filter.as_ref(), order, range, block)?;
```

Comment:
I don't fully understand why the `FilterCollection` needs to be built here and we therefore expose that detail. Couldn't we leave it as a detail in `FilterQuery::new`?

Reply:
Yes, we definitely could.  That was meant to be an optimization to pass in a FilterCollection ref instead of the entire Layout, but upon second thought I doubt the compiled programs differ significantly if at all.  
I see your point in containing the filter building logic within FilterQuery::new(); changed back.

Reply:
I moved FillterCollection new outside of FilterQuery::new() because I updated the collection property of FilterQuery to be a reference.  I couldn't make the compiler happy when creating the FilterCollection within FilterQuery: `returns a value referencing data owned by the current function`.